### PR TITLE
Fix sync errors after network issues

### PR DESCRIPTION
**Pull request by mejo-**
### 📝 Summary

* Resolves: #4340
* Resolves: #4414

### 🚧 TODO

- [x] Look into Cypress test
- [ ] Remove `.only` from Cypress test
- [x] remove `disableBc: true` from `SyncServiceProvider.js`

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation is not required

### CODE REVIEW

File: src/services/SyncServiceProvider.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ export default function createSyncServiceProvider({ ydoc, syncService, fileId, i
 		'ws://localhost:1234',
 		'file:' + fileId,
 		ydoc,
-		{ WebSocketPolyfill },
+		{ WebSocketPolyfill, disableBc: true },
```
Comment (juliushaertl) [inline src/services/SyncServiceProvider.js:45]: Nice, maybe something to still keep in debug mode?

Comment (max-nextcloud) [reply src/services/SyncServiceProvider.js:45]: I wonder how we could best trigger this from the dev environment.

**Comment by max-nextcloud** [comment]
I could not reproduce the cypress failure locally... set a crazy long wait time :tm: for the failed cypress statement to see if it's a matter of speed.

File: src/services/SyncServiceProvider.js (hunk 1 above)
Comment (mejo-) [reply src/services/SyncServiceProvider.js:45]: At another place we use `window._oc_debug` for this: https://github.com/nextcloud/text/blob/519637de1ccd75f9620060ff25b8f8f758da8a61/src/editor.js#L98

Comment (mejo-) [reply src/services/SyncServiceProvider.js]: I pushed a commit to set `disableBc` according to `window?._oc_debug`.

**Comment by max-nextcloud** [comment]
Ahh... now the `SyncServiceProvider.spec` is failing because the browser broadcast kicks in.

File: src/services/SyncServiceProvider.js (hunk 1 above)
Comment (mejo-) [reply src/services/SyncServiceProvider.js]: Interesting, this broke our `recovers from a dropped message` Cypress test.

**Comment by juliushaertl** [comment]
One minor unrelated UI bug, the notice about connection issues seems to no longer be sticky, but unrelated to this change:

**Comment by juliushaertl** [comment]
One minor unrelated UI bug, the notice about connection issues seems to no longer be sticky, but unrelated to this change:

Screenshot 2023-07-11 at 16 48 05

**Comment by mejo-** [comment]
> One minor unrelated UI bug, the notice about connection issues seems to no longer be sticky, but unrelated to this change:

Interesting, good catch. Let's open a new issue to track it: https://github.com/nextcloud/text/issues/4483

**Comment by mejo-** [comment]
/backport 6454257361d0ef158c04dab981fe949452449646,c012aac34c0a47f9b3fae67878a5244a9ac12638,2a666d1ea5958abb26f99cc39cd9372f5c802bc5,6a4489e8cce0d73bcee115fe472738cef180270a,155bbb5fa5783ef08e3ac8a6d3106c236d5c606e,0329f46f61420d5d4fbcf1c7da02d3cb71712161,58fc762c4da53ec6275513c447275b835371b67b,fa5d293a2df74b4a6189c3c4cd9f3bed41d47981,d196ae0d1fd16eb2a5ba2c08e400c92e4c7adb0f,f0396923a4ed15ec104d7e9533ac3a053cea725b,236b07ef71353083d2bafaac86bbe81973d3116e to stable27

**Comment by mejo-** [comment]
Manual backports:

- stable27: https://github.com/nextcloud/text/pull/4486
- stable26: https://github.com/nextcloud/text/pull/4487

**Pull request merged**